api/queries_sponsors_test.go

```diff
@@ -0,0 +1,118 @@
+package api
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_querySponsors(t *testing.T) {
+	var tests = []struct {
+		name           string
+		httpStubs      func(*httpmock.Registry)
+		expectedResult SponsorQuery
+		expectError    bool
+	}{
+		{
+			name: "success",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query SponsorsList`),
+					httpmock.StatusStringResponse(200, `{"data": {"user": {"sponsors": {"totalCount": 2, "nodes": [{"login": "mona"}, {"login": "lisa"}]}}}}`))
+			},
+			expectedResult: SponsorQuery{
+				User: User{
+					Sponsors: Sponsors{
+						TotalCount: 2,
+						Nodes: []Node{
+							{Login: "mona"},
+							{Login: "lisa"},
+						},
+					},
+				},
+			},
+			expectError: false,
+		},
+		{
+			name: "error",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query SponsorsList`),
+					httpmock.StatusStringResponse(500, `Internal Server Error`))
+			},
+			expectedResult: SponsorQuery{},
+			expectError:    true,
+		},
+	}
+
+	queryFnData := []struct {
+		name    string
+		queryFn func(*Client, string, string) (SponsorQuery, error)
+	}{
+		{
+			name:    "querySponsorsViaReflection",
+			queryFn: querySponsorsViaReflection,
+		},
+		{
+			name:    "querySponsorsViaStringManipulation",
+			queryFn: querySponsorsViaStringManipulation,
+		},
+	}
+
+	for _, tt := range tests {
+		for _, queryFn := range queryFnData {
+			t.Run(fmt.Sprintf("%s %s", tt.name, queryFn.name), func(t *testing.T) {
+				http := &httpmock.Registry{}
+				if tt.httpStubs != nil {
+					tt.httpStubs(http)
+				}
+
+				client := newTestClient(http)
+
+				sponsorsList, err := queryFn.queryFn(client, "github.com", "octocat")
+				if (err != nil) != tt.expectError {
```

Comment:
I'm absolutely sure this exists in a million places but I think it's too clever. I would much prefer to see:

```
if tt.expectError {
    require.Error(t, err)
}
```

Could be personal opinion.